src/huggingface_hub/_space_api.py

```diff
@@ -116,10 +116,10 @@ class SpaceRuntime:
 
     def __init__(self, data: Dict) -> None:
         self.stage = data["stage"]
-        self.hardware = data["hardware"]["current"]
-        self.requested_hardware = data["hardware"]["requested"]
-        self.sleep_time = data["gcTimeout"]
-        self.storage = data["storage"]
+        self.hardware = data.get("hardware", {}).get("current")
+        self.requested_hardware = data.get("hardware", {}).get("requested")
+        self.sleep_time = data.get("gcTimeout")
+        self.storage = data.get("storage")
```

Comment:
syntax is not super pretty but LGTM

Reply:
I'll yet you open an issue on https://github.com/python/cpython and in the meantime merge this PR :smile:

Reply:
@Wauplin : https://peps.python.org/pep-0505/#pep-deferral

Reply:
or https://discuss.python.org/t/introducing-a-safe-navigation-operator-in-python/35480/6